src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -590,8 +590,8 @@ private bool ShouldBeInStateSyncMode(Snapshot best)
             bool notInFastSync = !best.IsInFastSync;
             bool notNeedToWaitForHeaders = NotNeedToWaitForHeaders;
             bool stickyStateNodes = best.TargetBlock - best.Header < (FastSyncLag + StickyStateNodesDelta);
-            bool stateNotDownloadedYet = (best.TargetBlock - best.State > FastSyncLag ||
-                                          best.Header > best.State && best.Header > best.Block);
+            bool stateNotDownloadedYet = best.State <= 0 ||
+                                         (best.Header > best.State && best.Header > best.Block);
```

Comment:
So is the problem that `FastSyncLag` wasn't working properly? Or maybe the actual problem was the parenthesis were wrong? I expect the latter.

Reply:
FastSyncLag is 32; I missed 47k blocks while offline and in the 6 days following while it was in a limbo state of constantly switching to StateSync. Changing to `best.State <= 0` it was back online and fully synced in 2hrs.

Reply:
`FastSyncLag` shouldn't be `32` by any default

Reply:
Its a constant, was changed from `2` to `32` over a year ago https://github.com/NethermindEth/nethermind/pull/3944

Reply:
Issue is if you've been processing blocks; should it ever try to switch to state sync?

Reply:
I think what we would expect is `IsInAStickyFullSyncMode` to be true, wonder why it wasn't

Reply:
Think same scenario @MarekM25 was looking at (different area) maybe?
```csharp
private bool IsInAStickyFullSyncMode(Snapshot best)
{
    long bestBlock = Math.Max(best.Processed, LastBlockThatEnabledFullSync ?? 0);
    bool hasEverBeenInFullSync = bestBlock > _pivotNumber && best.State > _pivotNumber;
    long heightDelta = best.TargetBlock - bestBlock;
    return hasEverBeenInFullSync && heightDelta < FastSyncCatchUpHeightDelta;
}
```
Pivot update (new version or get latest); but head is behind the new pivot?

Reply:
Hm... Maybe then just change:

`bool hasEverBeenInFullSync = bestBlock > _pivotNumber && best.State > _pivotNumber;`

to

`bool hasEverBeenInFullSync = bestBlock > 0 && best.State > 0; ?